Kernel/Net/Routing.cpp

```diff
@@ -205,6 +205,10 @@ RoutingDecision route_to(const IPv4Address& target, const IPv4Address& source, c
         return { nullptr, {} };
     }
 
+    // If it's a broadcast, we already know everything we need to know.
+    if (target_addr == 0xffffffff && matches(adapter))
+        return { adapter, { 0xff, 0xff, 0xff, 0xff, 0xff, 0xff } };
```

Comment:
What should we do in case it's not a 255.255.255.255, but something like 192.168.1.255? So a broadcast, but not to *everyone* in the world, but just to a subnet?

Reply:
I'm not very clear on the software side of this, but I think we'd have to implement a broadcast routing algorithm for that (assuming that ARP fails and no one around us knows how to broadcast to a subnet), probably convert it to a series of unicasts.

Reply:
I'm nowhere near being an expert on this, but I would assume we should route such semi-broadcasts in much the same way, i.e. match the address against our adapters and then just send the broadcast there. No point in going through ARP since no individual peer has .255 anyway.

Reply:
There is no "there" in this case, the only reason we can easily route network-wide broadcasts is because there's a reserved MAC address (ff:ff:ff:ff:ff:ff) for it.
To broadcast to a subnet, our best bet is to find someone that can do that for us - hence ARP.

I'm not an expert either, so I might be very wrong either way.

Reply:
It's my understanding that when broadcasting to a subnet, we also use the `ff:ff:ff:ff:ff:ff` MAC (as in, broadcast to everyone connected to this cable!) with a proper broadcast IP (.255). That is of course if we're connected to the subnet ourselves, otherwise we'd have to route it as usual to our gateway.

Reply:
Hmmm...you might be right :thinking:
I'll slap a FIXME there, someone that knows about this more than I do can address that.